Read in the Hospitals dataset

In [ ]:
import pandas as pd
hospitals = pd.read_csv("../data/Hospitals.csv")

Look into the first few rows and last few rows

In [8]:
hospitals.head()

X             Y  OBJECTID        ID  \
0 -1.331889e+07  4.346975e+06         1   5793230   
1 -1.322651e+07  4.049626e+06         2  53391362   
2 -1.315620e+07  4.031978e+06         3  11190023   
3 -1.317190e+07  4.041752e+06         4  17090028   
4 -1.313208e+07  4.037270e+06         5  23691706   

                                                NAME  \
0                    CENTRAL VALLEY GENERAL HOSPITAL   
1  LOS ROBLES HOSPITAL & MEDICAL CENTER - EAST CA...   
2                  EAST LOS ANGELES DOCTORS HOSPITAL   
3          SOUTHERN CALIFORNIA HOSPITAL AT HOLLYWOOD   
4                      KINDRED HOSPITAL BALDWIN PARK   

                     ADDRESS             CITY STATE    ZIP           ZIP4  \
0    1025 NORTH DOUTY STREET          HANFORD    CA  93230  NOT AVAILABLE   
1             150 VIA MERIDA  WESTLAKE VILAGE    CA  91362  NOT AVAILABLE   
2    4060 WHITTIER BOULEVARD      LOS ANGELES    CA  90023  NOT AVAILABLE   
3     6245 DE LONGPRE AVENUE        HOLLYWOOD    CA  90028  NOT AVAILABLE   
4  14148 FRANCISQUITO AVENUE     BALDWIN PARK    CA  91706  NOT AVAILABLE   

   ...             VAL_DATE                           WEBSITE       STATE_ID  \
0  ...  2014/02/10 00:00:00      http://www.hanfordhealth.com  NOT AVAILABLE   
1  ...  2014/02/10 00:00:00  http://www.losrobleshospital.com  NOT AVAILABLE   
2  ...  2014/02/10 00:00:00             http://www.elalax.com  NOT AVAILABLE   
3  ...  2014/02/10 00:00:00         http://sch-hollywood.com/  NOT AVAILABLE   
4  ...  2014/02/10 00:00:00      http://www.khbaldwinpark.com  NOT AVAILABLE   

                                    ALT_NAME ST_FIPS        OWNER TTL_STAFF  \
0                              NOT AVAILABLE       6  PROPRIETARY      -999   
1                              NOT AVAILABLE       6  PROPRIETARY      -999   
2                              NOT AVAILABLE       6  PROPRIETARY      -999   
3  HOLLYWOOD COMMUNITY HOSPITAL OF HOLLYWOOD       6  PROPRIETARY      -999   
4                              NOT AVAILABLE       6  PROPRIETARY      -999   

   BEDS         TRAUMA  HELIPAD  
0    49  NOT AVAILABLE        N  
1    62  NOT AVAILABLE        N  
2   127  NOT AVAILABLE        N  
3   100  NOT AVAILABLE        N  
4    95  NOT AVAILABLE        N  

[5 rows x 34 columns]

In [9]:
hospitals.tail()

X             Y  OBJECTID         ID  \
7591 -8.566905e+06  4.728815e+06      7607  183520904   
7592 -1.219624e+07  3.676931e+06      7608  192185607   
7593 -8.084677e+06  5.094112e+06      7609  196706457   
7594 -8.522423e+06  4.718512e+06      7610  191021401   
7595 -1.244921e+07  4.976468e+06      7611  196784108   

                                               NAME  \
7591  ADVENTIST HEALTHCARE WHITE OAK MEDICAL CENTER   
7592      COPPER QUEEN DOUGLAS EMERGENCY DEPARTMENT   
7593                      WHITING FORENSIC HOSPITAL   
7594             J KENT MCNEW FAMILY MEDICAL CENTER   
7595                  UNIVERSITY ORTHOPAEDIC CENTER   

                         ADDRESS            CITY STATE    ZIP           ZIP4  \
7591           11890 HEALING WAY   SILVER SPRING    MD  20904  NOT AVAILABLE   
7592           100 E. 5TH STREET         DOUGLAS    AZ  85607  NOT AVAILABLE   
7593                70 OBRIEN DR      MIDDLETOWN    CT   6457           3945   
7594  175 HARRY S TRUMAN PARKWAY       ANNAPOLIS    MD  21401  NOT AVAILABLE   
7595              590 WAKARA WAY  SALT LAKE CITY    UT  84108  NOT AVAILABLE   

      ...             VAL_DATE  \
7591  ...  2020/05/05 00:00:00   
7592  ...  2020/05/05 00:00:00   
7593  ...  2020/05/08 00:00:00   
7594  ...  2020/05/05 00:00:00   
7595  ...  2020/05/05 00:00:00   

                                                WEBSITE       STATE_ID  \
7591  https://www.adventisthealthcare.com/locations/...  NOT AVAILABLE   
7592  https://cqch.org/cqch-clinics/copper-queen-dou...  NOT AVAILABLE   
7593  https://portal.ct.gov/dmhas/wfh/whiting-forens...  NOT AVAILABLE   
7594  https://aahs.org/locations/j--kent-mcnew-famil...  NOT AVAILABLE   
7595          https://healthcare.utah.edu/orthopaedics/  NOT AVAILABLE   

           ALT_NAME ST_FIPS               OWNER TTL_STAFF  BEDS  \
7591  NOT AVAILABLE      24          NON-PROFIT      -999   178   
7592  NOT AVAILABLE       4          NON-PROFIT      -999  -999   
7593  NOT AVAILABLE       9  GOVERNMENT - STATE      -999   229   
7594  NOT AVAILABLE      24          NON-PROFIT      -999  -999   
7595  NOT AVAILABLE      49  GOVERNMENT - STATE      -999  -999   

             TRAUMA  HELIPAD  
7591  NOT AVAILABLE        N  
7592       LEVEL IV        N  
7593  NOT AVAILABLE        N  
7594  NOT AVAILABLE        N  
7595  NOT AVAILABLE        N  

[5 rows x 34 columns]

By looking at the first few rows and last few rows we can see that the dataset has information on hospitals from all over the United States

In [1]:
hospitals.shape

NameError: name 'hospitals' is not defined

The hospitals dataframe has 7597 rows and 34 columns

In [13]:
hospitals = hospitals[['NAME', 'ADDRESS', 'CITY', 'STATE', 'TYPE',  'POPULATION', 'COUNTY', 'COUNTYFIPS', 'NAICS_DESC', 'BEDS']]

In [15]:
hospital_types = hospitals["TYPE"].unique()

In [16]:
print(hospital_types)

['GENERAL ACUTE CARE' 'PSYCHIATRIC' 'CHILDREN' 'LONG TERM CARE'
 'CRITICAL ACCESS' 'REHABILITATION' 'MILITARY' 'WOMEN' 'SPECIAL'
 'CHRONIC DISEASE']


The hospital types are: 'GENERAL ACUTE CARE' 'PSYCHIATRIC' 'CHILDREN' 'LONG TERM CARE'
 'CRITICAL ACCESS' 'REHABILITATION' 'MILITARY' 'WOMEN' 'SPECIAL'
 'CHRONIC DISEASE'

In [18]:
hospitals["TYPE"].value_counts()

GENERAL ACUTE CARE    4481
CRITICAL ACCESS       1027
PSYCHIATRIC            759
LONG TERM CARE         431
REHABILITATION         378
MILITARY               216
SPECIAL                145
CHILDREN               130
WOMEN                   20
CHRONIC DISEASE          9
Name: TYPE, dtype: int64

General Acute Care is the most common type of hospital

In [22]:
nash_hospitals = hospitals.loc[hospitals["CITY"]=="NASHVILLE"]

In [24]:
print(nash_hospitals["BEDS"])

545      213
2083      63
3449      70
3485      60
3780      50
4253      20
5429    -999
5433     150
6289     683
6293      80
6661     493
6774      23
6777     541
6786     126
6808     733
6813    1091
Name: BEDS, dtype: int64


One of the beds values is -999. This probably indicates that no value exists for "BEDS" for that particular hospital.

In [29]:
tenn_hospitals = hospitals.loc[hospitals["STATE"]=="TN"]

In [30]:
tenn_hospitals["COUNTY"].value_counts()

SHELBY        23
DAVIDSON      15
HAMILTON      13
KNOX          11
WASHINGTON     6
              ..
HAWKINS        1
OBION          1
HENRY          1
CLAY           1
JEFFERSON      1
Name: COUNTY, Length: 80, dtype: int64

Shelby county has the most hospitals with 23

In [32]:
tenn_hospitals["COUNTY"].nunique()

80

There are 80 unique counties listed in the Tennessee hospitals database. Therefor, 15 counties have no hospitals at all.